src/components/encounter-list/encounter-list.component.tsx

```diff
@@ -161,79 +161,35 @@ const EncounterList: React.FC<EncounterListProps> = ({
         row[column.key] = val;
       });
 
-      //TODO: This 'piece of hack' SHOULD BE REMOVED once Pius is done with the Actions implementation
-      if (form.package == 'covid' && form.name == 'covid_assessment') {
+      if (row['actions']) {
+        let actionItems = row['actions'];
+        delete row['actions'];
         row['actions'] = (
           <OverflowMenu flipped className={styles.flippedOverflowMenu}>
-            <OverflowMenuItem
-              itemText={'View Case'}
-              onClick={e => {
-                e.preventDefault();
-                launchFormInViewMode(
-                  applyFormIntent('', getForm('covid', 'covid_case')),
-                  encounter.uuid,
-                  forceComponentUpdate,
-                );
-              }}
-            />
-            <OverflowMenuItem
-              itemText={'Edit Case'}
-              onClick={e => {
-                e.preventDefault();
-                launchFormInEditMode(
-                  applyFormIntent('', getForm('covid', 'covid_case')),
-                  encounter.uuid,
-                  forceComponentUpdate,
-                );
-              }}
-            />
-            <OverflowMenuItem
-              itemText={'Edit Outcome'}
-              onClick={e => {
-                e.preventDefault();
-                launchFormInEditMode(
-                  applyFormIntent('', getForm('covid', 'covid_outcome')),
-                  encounter.uuid,
-                  forceComponentUpdate,
-                );
-              }}
-            />
+            {actionItems.map((actionItem, index) => (
+              <OverflowMenuItem
+                itemText={actionItem.label}
+                onClick={e => {
+                  e.preventDefault();
+                  if (actionItem.mode == 'edit') {
+                    launchFormInEditMode(
+                      applyFormIntent(actionItem.intent, getForm(actionItem.form.package, actionItem.form.name)),
+                      actionItem.encounterUuid,
+                      forceComponentUpdate,
+                    );
+                  } else {
```

Comment:
Don't we have a case were we may want to launch a form in `enter` mode?

Reply:
Am not sure about this for encounter actions. Will look into it to be sure